kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -647,14 +920,38 @@ compareForEquals first second
     | isConstructorLike second = GT
     | otherwise = compare first second
 
-bytesDifferent ::
-    MonadUnify unifier =>
+{- | Matches
+
+@
+\\equals{_, _}(\\dv{Bytes}(bytes1), \\dv{Bytes}(bytes2))
+@
+
+and
+
+@
+\\and{_}(\\dv{Bytes}(bytes1), \\dv{Bytes}(bytes2))
+@
+
+when
+
+@
+bytes1 /= bytes2
+@
+-}
+matchBytesDifferent ::
     TermLike RewritingVariableName ->
     TermLike RewritingVariableName ->
-    MaybeT unifier (Pattern RewritingVariableName)
-bytesDifferent
-    (Recursive.project -> _ :< InternalBytesF (Const bytesFirst))
-    (Recursive.project -> _ :< InternalBytesF (Const bytesSecond))
-        | bytesFirst /= bytesSecond =
-            return Pattern.bottom
-bytesDifferent _ _ = empty
+    Maybe ()
+matchBytesDifferent first second
+    | _ :< InternalBytesF (Const bytesFirst) <- Recursive.project first
+      , _ :< InternalBytesF (Const bytesSecond) <- Recursive.project second
+      , bytesFirst /= bytesSecond =
+        Just ()
+    | otherwise = Nothing
+{-# INLINE matchBytesDifferent #-}
+
+bytesDifferent ::
+    MonadUnify unifier =>
+    unifier (Pattern RewritingVariableName)
+bytesDifferent =
+    return Pattern.bottom
```

Comment:
```suggestion
matchBytesDifferent first second
    | _ :< InternalBytesF (Const bytesFirst) <- Recursive.project first
      , _ :< InternalBytesF (Const bytesSecond) <- Recursive.project second =
        Just ()
    | otherwise = Nothing
{-# INLINE matchBytesDifferent #-}

bytesDifferent ::
    MonadUnify unifier =>
    unifier (Pattern RewritingVariableName)
bytesDifferent UnifyBytes{ bytesFirst, bytesSecond }
    | bytesFirst == bytesSecond =
        return $ Pattern.fromTermLike $ internalBytes bytesFirst
    | otherwise =
        empty
```